coalib/settings/SectionFilling.py

```diff
@@ -25,17 +26,38 @@ def fill_settings(sections, acquire_settings, log_printer):
     local_bears = {}
     global_bears = {}
 
+    bears_by_section = {}
+    bear_dirs_by_section = {}
+
     for section_name, section in sections.items():
-        bear_dirs = section.bear_dirs()
-        bears = list(section.get("bears", ""))
-        section_local_bears, section_global_bears = collect_bears(
-            bear_dirs,
-            bears,
-            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
-            log_printer)
-        all_bears = copy.deepcopy(section_local_bears)
-        all_bears.extend(section_global_bears)
-        fill_section(section, acquire_settings, log_printer, all_bears)
+        bear_dirs_by_section[section_name] = section.bear_dirs()
+        bears_by_section[section_name] = list(section.get("bears", ""))
+
+    bears_to_collect = set(chain.from_iterable(bears_by_section.values()))
+    bear_dirs_to_collect = set(chain(*bear_dirs_by_section.values()))
```

Comment:
I confess I did this for the sake of line length. Hobgoblin of Little Minds
and all that...

On 10 Aug 2016 08:35, "Lasse Schuirmann" <notifications@github.com> wrote:

> In coalib/settings/SectionFilling.py
> <https://github.com/coala-analyzer/coala/pull/2594#discussion_r74198024>:
>
> >      for section_name, section in sections.items():
> > -        bear_dirs = section.bear_dirs()
> > -        bears = list(section.get("bears", ""))
> > -        section_local_bears, section_global_bears = collect_bears(
> > -            bear_dirs,
> > -            bears,
> > -            [BEAR_KIND.LOCAL, BEAR_KIND.GLOBAL],
> > -            log_printer)
> > -        all_bears = copy.deepcopy(section_local_bears)
> > -        all_bears.extend(section_global_bears)
> > -        fill_section(section, acquire_settings, log_printer, all_bears)
> > +        bear_dirs_by_section[section_name] = section.bear_dirs()
> > +        bears_by_section[section_name] = list(section.get("bears", ""))
> > +
> > +    bears_to_collect = set(chain.from_iterable(bears_by_section.values()))
> > +    bear_dirs_to_collect = set(chain(*bear_dirs_by_section.values()))
>
> first you use chain.from_iterable and then chain with *, just use one of
> them consistenly shall we? ;)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/coala-analyzer/coala/pull/2594/files/d33f368876807bf83487f02330eeb47dfae37c24#r74198024>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADrZ-M4IaROnb8BzfIoCAZZ673m32sPpks5qeX9YgaJpZM4JfA00>
> .
>